client/request/request.go

```diff
@@ -20,6 +20,41 @@ type Request struct {
 	Subscription []*OperationDefinition
 }
 
+// Operation returns the operation that should be executed from the request.
+//
+// The operationName is used to select an operation when multiple are defined.
+func (r *Request) Operation(operationName string) (*OperationDefinition, error) {
+	switch {
+	case len(r.Queries) == 1 && len(r.Mutations) == 0 && len(r.Subscription) == 0:
+		return r.Queries[0], nil
+
+	case len(r.Queries) == 0 && len(r.Mutations) == 1 && len(r.Subscription) == 0:
+		return r.Mutations[0], nil
+
+	case len(r.Queries) == 0 && len(r.Mutations) == 0 && len(r.Subscription) == 1:
+		return r.Subscription[0], nil
+
+	case len(r.Queries) == 0 && len(r.Mutations) == 0 && len(r.Subscription) == 0:
+		return nil, ErrMissingQueryOrMutation
+	}
+	for _, op := range r.Queries {
+		if op.Name == operationName {
+			return op, nil
+		}
+	}
+	for _, op := range r.Mutations {
+		if op.Name == operationName {
+			return op, nil
+		}
+	}
+	for _, op := range r.Subscription {
+		if op.Name == operationName {
+			return op, nil
```

Comment:
suggestion: Similar to above mutation comment test in combination with `mutation` and `query`. However if this is too much trouble don't bother.